### ISSUE TITLE
No documentation/tutorials

### ISSUE THREAD

**Issue by kitkat14** [tommcfarlin/WordPress-Plugin-Boilerplate#269]
First of all i want to thank you for making such a great plugin in order that people will start making plugins as they should have (including me).

But your mission will not really succeed if people dont know how to use it... I have waited some months but  there are still nothing happening on that front...

So any plans to make 1 tutorial of how to use it to create backend and a front end stuff from the boilerplate ?

**Comment by tommcfarlin** [comment]
> But your mission will not really succeed if people dont know how to use it... I have waited some months but there are still nothing happening on that front...

I know - documentation is something that I've wanted to complete for sometime, but given that this is a free project and is something that I can work on only as time allows, I simply haven't had time to work on it just yet.

I do have a course coming out on how to use it to build a plugin soon and I'll blog about it as soon as it's released.

I appreciate the patience.

**Comment by kitkat14** [comment]
Yeah I know :) it's free and in your own time so it's double appreciated. Where do you have courses ? Udemy ?

**Comment by tommcfarlin** [comment]
> Where do you have courses? 

No - this one will be featured on WP Sessions sometime in the near future :).

**Comment by kitkat14** [comment]
Oke :)

Maybe some qq you can answer.

The public folder is all that happens on the front end right ?
Say I want to add a class Building and add bootstrap CSS and JS. You make a
class in that folder and add the stuff in the corresponding CSS and JS
folder right.

So how do you enque more than one js in the public php class ? I am getting
a bit confused with so much separation :)

Op dinsdag 24 februari 2015 heeft Tom McFarlin <notifications@github.com>
het volgende geschreven:

**Comment by tommcfarlin** [comment]
> The public folder is all that happens on the front end right ?

That's correct.

> Say I want to add a class Building and add bootstrap CSS and JS. You make a
class in that folder and add the stuff in the corresponding CSS and JS
folder right.

Assuming that you want to use Bootstrap in the front-end, then you'd place those in the `public/js` and `public/css` folders; however, since they are libraries, I'd also consider creating a `lib` directory in each of the subdirectories to indicate that these are third-party files.

Secondly, I'd also do a check in the code to make sure Bootstrap hasn't already been registered and enqueued since it's such a popular framework.

> So how do you enque more than one js in the public php class ?

You'd make multiple calls to `wp_enqueue_script` - one for each file that you want to enqueue. Just make sure that you order them properly and that you set the proper dependencies so they are loaded correctly on the front-end.

**Comment by kitkat14** [comment]
But in the public class you do something like this.

Public function enqueue_styles(){
wp_enqueue_style($this-classname, and so on)
}

If you do this a second time (so you want to add another script or CSS) it
will not work.

Op woensdag 25 februari 2015 heeft Tom McFarlin <notifications@github.com>
het volgende geschreven:

**Comment by tommcfarlin** [comment]
> If you do this a second time (so you want to add another script or CSS) it will not work.

It won't work if you don't give each script a proper ID. If you make sure they are unique. See [this page](http://codex.wordpress.org/Function_Reference/wp_enqueue_script) for more information.

**Comment by user52839307** [comment]
Thank you for clearing that up. I could not understand why it would not load more than one script, it was staring me in the face. 

Your examples for loading the javascript and css are using the variable `$this->plugin_name` for the handle. I just needed to add a unique extension string like `$this->plugin_name . '-javascript' ` for each additional script and it worked fine.

**Comment by Nebri** [comment]
I have found this tutorial to be extremely useful when getting started with this plugin:

https://www.sitepoint.com/wordpress-plugin-boilerplate/
